spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/ArgumentSanitizer.java

```diff
@@ -16,17 +16,30 @@
 
 package org.springframework.cloud.dataflow.server.controller.support;
 
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import org.springframework.util.StringUtils;
+
 /**
  * Sanitizes potentially sensitive keys for a specific command line arg.
+ *
  * @author Glenn Renfro
  */
 public class ArgumentSanitizer {
-	private static final String[] REGEX_PARTS = { "*", "$", "^", "+" };
+	private static final String[] REGEX_PARTS = {"*", "$", "^", "+"};
+
+	private static final String REDACTION_STRING = "******";
+
+	private static final String[] KEYS_TO_SANITIZE = {"password", "secret", "key", "token", ".*credentials.*",
+			"vcap_services"};
+
+	public static Pattern passwordParameterPatternForStreams = Pattern.compile(
+			"(?i)(--[\\p{Z}]*[\\p{L}]*("
+					+ StringUtils.arrayToDelimitedString(KEYS_TO_SANITIZE, "|")
+					+ ")[\\p{L}]*[\\p{Z}]*=[\\p{Z}]*)((\"[\\p{L}|\\p{Pd}|\\p{Ps}|\\p{Pe}|\\p{Pc}|\\p{S}|\\p{N}|\\p{Z}]*\")|([\\p{N}|\\p{L}|\\p{Po}|\\p{Pc}|\\p{S}]*))",
+			Pattern.UNICODE_CASE);
```

Comment:
A small comment on what that regex actually does would be helpful for "the next guy".